dowhy/causal_refuter.py

```diff
@@ -11,19 +12,94 @@ class CausalRefuter:
     """
     # Default value for the number of simulations to be conducted
     DEFAULT_NUM_SIMULATIONS = 100
+    
+    def sanity_check_required_variables(self, required_variables):
```

Comment:
This is a strange function. It either changes the value of self.invert, throws an exception or does nothing. 
Ideally, functions should do one thing, and do that well. 

In this case, here's my suggestion: 
## Combine sanity_check and choose_variables
Then the new choose_variables function returns the final variable names. This is nice because it has a simple contract--give me required_variables parameter and I will return the list of required variables.  It also makes it possible to use invert as a simple local variable. No need to create an instance attribute that is only concerned with processing a parameter, and is useless afterwards.
Finally, code-wise too, it removes multiple if-statements checking "required_variables" and you can easily merge the current choose_variables code in the right if-condition.